wpimath/src/test/java/edu/wpi/first/math/WPIMathJNITest.java

```diff
@@ -6,11 +6,58 @@
 
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Twist3d;
 import org.junit.jupiter.api.Test;
 
 public class WPIMathJNITest {
   @Test
   public void testLink() {
     assertDoesNotThrow(WPIMathJNI::forceLoad);
   }
+
+  public static double timePose3dExpMillis(Pose3d start, Twist3d twist, int count) {
+    final long startTime = System.nanoTime();
+    for (int i = 0; i < count; i++) {
+      start.exp(twist);
+    }
+    final long endTime = System.nanoTime();
+    double totalTimeMillis = (endTime - startTime) / 1000.0;
+    return totalTimeMillis / count;
+  }
+
+  public static double timePose3dLogMillis(Pose3d start, Pose3d end, int count) {
+    final long startTime = System.nanoTime();
+    for (int i = 0; i < count; i++) {
+      start.log(end);
+    }
+    final long endTime = System.nanoTime();
+    double totalTimeMillis = (endTime - startTime) / 1000.0;
```

Comment:
endTime and startTime are in nanoseconds, so dividing by 1000 gives microseconds, not milliseconds.